worker/caasmodelconfigmanager/worker.go

```diff
@@ -107,13 +126,52 @@ func (w *manager) Wait() error {
 	return w.catacomb.Wait()
 }
 
-func (w *manager) loop() error {
+func (w *manager) loop() (err error) {
 	controllerConfig, err := w.config.Facade.ControllerConfig()
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if err := w.config.Broker.EnsureImageRepoSecret(controllerConfig.CAASImageRepo()); err != nil {
+	repoDetails := controllerConfig.CAASImageRepo()
+	if !repoDetails.IsPrivate() {
+		// No ops for public registry config.
+		return nil
+	}
+	w.reg, err = w.registryFunc(repoDetails)
+	if err != nil {
 		return errors.Trace(err)
 	}
-	return nil
+	err = w.reg.Ping()
+	if utils.IsPublicAPINotAvailableError(err) {
+		// No ops for public registry config.
+		// This should never happen because we have done this check above.
+		return nil
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if err := w.ensureImageRepoSecret(true); err != nil {
+		return errors.Trace(err)
+	}
+	for {
+		select {
+		case <-w.catacomb.Dying():
+			return w.catacomb.ErrDying()
+		case <-w.clock.After(3 * time.Second):
```

Comment:
3 seconds seems too short?

Reply:
Yeah, Changed to let the `ShouldRefreshAuth` method to decide the next tick;